packages/datadog-plugin-kafkajs/src/producer.js

```diff
@@ -31,8 +28,14 @@ class KafkajsProducerPlugin extends ProducerPlugin {
     }
     for (const message of messages) {
       if (typeof message === 'object') {
-        if (this.config.dsmEnabled) message.headers['dd-pathway-ctx'] = pathwayCtx
         this.tracer.inject(span, 'text_map', message.headers)
+        if (this.config.dsmEnabled) {
+          const payloadSize = getMessageSize(message)
+          const dataStreamsContext = this.tracer
+            .setCheckpoint(['direction:out', `topic:${topic}`, 'type:kafka'], payloadSize)
+          pathwayCtx = encodePathwayContext(dataStreamsContext)
+          message.headers[CONTEXT_PROPAGATION_KEY] = pathwayCtx
```

Comment:
I see how you need to to this after getting the `dataStreamsContext`, but changing the headers after computing the size means we're going to underestimate it by a little bit :/

Reply:
I dont believe this to be true, the `CONTEXT_PROPAGATION_KEY ` is a constant and `pathwayCtx ` is always a 20 byte buffer.

Reply:
ah do you mean that the key/value is already present in the message before and this is merely updating it ?
I though the key was inserted here.

Reply:
See below